File: mobile/lib/core/models/application.dart
enum ApplicationStatus { pending, submitted, failed, skipped }

class JobApplication {
  final int id;
  final int userId;
  final int? profileId;
  final String? jobTitle;
  final String? companyName;
  final String jobUrl;
  final String? jobDescription;
  final String? tailoredSummary;
  final List<String>? tailoredBullets;
  final List<String>? tailoredSkills;
  final int? matchScore;
  final ApplicationStatus status;
  final String? errorMessage;
  final DateTime? appliedAt;
  final DateTime createdAt;

  const JobApplication({
    required this.id,
    required this.userId,
    this.profileId,
    this.jobTitle,
    this.companyName,
    required this.jobUrl,
    this.jobDescription,
    this.tailoredSummary,
    this.tailoredBullets,
    this.tailoredSkills,
    this.matchScore,
    required this.status,
    this.errorMessage,
    this.appliedAt,
    required this.createdAt,
  });

  factory JobApplication.fromJson(Map<String, dynamic> json) {
    return JobApplication(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      profileId: json['profile_id'] as int?,
      jobTitle: json['job_title'] as String?,
      companyName: json['company_name'] as String?,
      jobUrl: json['job_url'] as String,
      jobDescription: json['job_description'] as String?,
      tailoredSummary: json['tailored_summary'] as String?,
      tailoredBullets: (json['tailored_bullets'] as List<dynamic>?)?.cast<String>(),
      tailoredSkills: (json['tailored_skills'] as List<dynamic>?)?.cast<String>(),
      matchScore: json['match_score'] as int?,
      status: ApplicationStatus.values.firstWhere(
        (e) => e.name == json['status'],
        orElse: () => ApplicationStatus.pending,
      ),
      errorMessage: json['error_message'] as String?,
      appliedAt: json['applied_at'] != null
          ? DateTime.parse(json['applied_at'] as String)
          : null,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  String get statusLabel {
    switch (status) {
      case ApplicationStatus.submitted:
        return 'Submitted';
      case ApplicationStatus.pending:
        return 'Pending';
      case ApplicationStatus.failed:
        return 'Failed';
      case ApplicationStatus.skipped:
        return 'Skipped';
    }
  }
}

File: backend/models/application.py
from datetime import datetime, timezone
from typing import Optional
from sqlalchemy import String, Text, DateTime, ForeignKey, JSON, Enum
from sqlalchemy.orm import Mapped, mapped_column, relationship
import enum

from backend.db.database import Base


class ApplicationStatus(str, enum.Enum):
    PENDING = "pending"
    SUBMITTED = "submitted"
    FAILED = "failed"
    SKIPPED = "skipped"


class JobApplication(Base):
    __tablename__ = "job_applications"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False, index=True)
    profile_id: Mapped[Optional[int]] = mapped_column(ForeignKey("profiles.id"), nullable=True)

    # Job info
    job_title: Mapped[Optional[str]] = mapped_column(String(255))
    company_name: Mapped[Optional[str]] = mapped_column(String(255))
    job_url: Mapped[str] = mapped_column(String(1000), nullable=False)
    job_description: Mapped[Optional[str]] = mapped_column(Text)

    # AI output
    tailored_summary: Mapped[Optional[str]] = mapped_column(Text)
    tailored_bullets: Mapped[Optional[list]] = mapped_column(JSON)
    tailored_skills: Mapped[Optional[list]] = mapped_column(JSON)
    match_score: Mapped[Optional[int]] = mapped_column()

    # Status
    status: Mapped[ApplicationStatus] = mapped_column(
        Enum(ApplicationStatus), default=ApplicationStatus.PENDING
    )
    error_message: Mapped[Optional[str]] = mapped_column(Text)

    applied_at: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True))
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )

    user: Mapped["User"] = relationship("User", back_populates="applications")  # noqa: F821
    profile: Mapped[Optional["Profile"]] = relationship(  # noqa: F821
        "Profile", back_populates="applications"
    )
